README.md

```diff
@@ -1,17 +1,110 @@
+│
+├── sdk-api/                       # API-specific integrations
+│   └── src/main/kotlin/io/etip/sdk/api/
+│       ├── config/                 # ClientConfig, Environment enum
+│       ├── integration/            # ApiService interface and implementation
+│       └── factory/                # SdkFactory to instantiate the API client
+│
+├── examples/                      # Example usage of the SDK
+│   └── basic-example/             # Pokémon API usage demo
+│
+├── .github/workflows/             # CI/CD workflows for release
+│   └── release.yml
+│
+└── README.md                      # Project overview and usage guide
+```
+
+### 🔐 Authentication Strategy
+
+The SDK follows a flexible authentication strategy that separates concerns between `sdk-core` and `sdk-api`.
+
+#### In `sdk-core`
+
+Authentication must be configured at the `HttpClient` level and is agnostic to the specific authentication mechanism. Whether the integration requires **Basic Auth**, **JWT Bearer tokens**, or any other custom scheme, the core does **not** hard-code any implementation. Instead, it provides the infrastructure (like `BaseHttpClient`) to accept headers, allowing the `sdk-api` layer to inject credentials dynamically.
+
+You can also implement custom authenticators using a strategy pattern by defining an abstract `Authenticator` interface in `sdk-core`. The concrete implementation (e.g., `JwtAuthenticator`, `BasicAuthenticator`) would live in `sdk-api`.
+
+#### In `sdk-api`
+
+Each API integration defines its own `ClientConfig` class to pass the necessary authentication details (e.g., username/password or token), base URL, environment, and additional settings.
+
+A `SdkFactory` is used to initialize the SDK with these configurations, build the proper `HttpClient` through `HttpClientFactory`, and return an instance of the `ApiService`.
+
+For example, in the **Pokémon API integration** included in the template, `ClientConfig` receives only the base URL, and no authentication is applied. In a real-world integration, you'd adjust this to support the target API’s authentication scheme.
+
+#### Example of basic authentication using Ktor Auth library
+
+This is the implementation of the HttpClientFactory implementing the Basic auth strategy:
+
+```
+object HttpClientFactory {
+    fun create(username: String, password: String): HttpClient {
+        return HttpClient(CIO) {
+            install(ContentNegotiation) {
+                json(Json {
+                    ignoreUnknownKeys = true
+                })
+            }
+
+            install(Logging) {
+                logger = Logger.DEFAULT
+                level = LogLevel.INFO
+            }
+
+            install(Auth) {
+                basic {
+                    credentials {
+                        BasicAuthCredentials(username = username, password = password)
+                    }
+                    sendWithoutRequest { true }
+                }
+            }
+        }
```

Comment:
How to handle if the external API requires a API Key or something in the URL instead of standard HTTP header?